src/rust/src/x509/certificate.rs

```diff
@@ -412,6 +412,17 @@ fn parse_display_text(
         DisplayText::IA5String(o) => Ok(pyo3::types::PyString::new(py, o.as_str()).to_object(py)),
         DisplayText::Utf8String(o) => Ok(pyo3::types::PyString::new(py, o.as_str()).to_object(py)),
         DisplayText::VisibleString(o) => {
+            if !UnvalidatedVisibleString::verify(o.as_str()) {
+                let cryptography_warning = py
+                    .import(pyo3::intern!(py, "cryptography.utils"))?
+                    .getattr(pyo3::intern!(py, "DeprecatedIn41"))?;
+                pyo3::PyErr::warn(
+                    py,
+                    cryptography_warning,
+                    "Invalid ASN.1 (UTF-8 characters in a VisibleString). A future version of cryptography will error on this input.",
```

Comment:
Can we provide more context on where the visible string is in the structure? I'd also suggest the second sentence be phrased more like "In a future version of cryptography, an exception will be raised in this circumstance."